src/main/java/graphql/language/AstHelper.java

```diff
@@ -0,0 +1,178 @@
+package graphql.language;
+
+import graphql.AssertException;
+import graphql.Scalars;
+import graphql.schema.GraphQLEnumType;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputType;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLNonNull;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLType;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AstHelper {
+
+    /*
+     * Produces a GraphQL Value AST given a Java value.
+     *
+     * A GraphQL type must be provided, which will be used to interpret different
+     * Java values.
+     *
+     * |      Value    | GraphQL Value        |
+     * | ------------- | -------------------- |
+     * | Object        | Input Object         |
+     * | Array         | List                 |
+     * | Boolean       | Boolean              |
+     * | String        | String / Enum Value  |
+     * | Number        | Int / Float          |
+     * | Mixed         | Enum Value           |
+     */
+    public static Value astFromValue(Object _value, GraphQLType type) {
```

Comment:
sure - funny thing is its a direct port of the JS implementation.  So its "reference mess" ;)

I agree its messy